ethereum/core/src/main/java/tech/pegasys/pantheon/ethereum/vm/operations/RevertOperation.java

```diff
@@ -36,8 +40,10 @@ public Gas cost(final MessageFrame frame) {
   public void execute(final MessageFrame frame) {
     final UInt256 from = frame.popStackItem().asUInt256();
     final UInt256 length = frame.popStackItem().asUInt256();
-
-    frame.setOutputData(frame.readMemory(from, length));
+    BytesValue reason = frame.readMemory(from, length);
+    frame.setOutputData(reason);
+    String reasonMessage = new String(reason.extractArray(), CHARSET);
```

Comment:
Strings in solidity are UTF-8 (https://solidity.readthedocs.io/en/v0.5.3/types.html#dynamically-sized-byte-array) which is where these values will come from.  Use `java.nio.charset.StandardCharsets.UTF_8`.

Reply:
Switched to UTF-8